Description: Fix C/C++ confusion due to GLU/autoconf issue.
Author: Paul Wise <pabs@debian.org>
Bug-Debian: https://bugs.debian.org/993870
Forwarded: not-needed
--- a/build/m4/ax_check_glu.m4
+++ b/build/m4/ax_check_glu.m4
@@ -16,7 +16,6 @@
 
 AC_DEFUN([AX_CHECK_GLU],
 [AC_REQUIRE([AX_CHECK_GL])dnl
-AC_REQUIRE([AC_PROG_CXX])dnl
 GLU_CFLAGS="${GL_CFLAGS}"
 if test "X${with_apple_opengl_framework}" != "Xyes"; then
   AC_CACHE_CHECK([for OpenGL Utility library], [ax_cv_check_glu_libglu],
@@ -33,16 +32,6 @@
       ax_try_lib="${ax_lib}"
     fi
     LIBS="${ax_try_lib} ${GL_LIBS} ${ax_save_LIBS}"
-    #
-    # libGLU typically links with libstdc++ on POSIX platforms. However,
-    # setting the language to C++ means that test program source is named
-    # "conftest.cc"; and Microsoft cl doesn't know what to do with such a
-    # file.
-    #
-    AC_LANG_PUSH([C++])
-    if test X$ax_compiler_ms = Xyes; then
-      AC_LANG_PUSH([C])
-    fi
     AC_LINK_IFELSE(
     [AC_LANG_PROGRAM([[
 # if HAVE_WINDOWS_H && defined(_WIN32)
@@ -51,10 +40,6 @@
 # include <GL/glu.h>]],
                      [[gluBeginCurve(0)]])],
     [ax_cv_check_glu_libglu="${ax_try_lib}"; break])
-    if test X$ax_compiler_ms = Xyes; then
-      AC_LANG_POP([C])
-    fi
-    AC_LANG_POP([C++])
   done
   LIBS=${ax_save_LIBS}
   CPPFLAGS=${ax_save_CPPFLAGS}])
